xhr/client/src/main/scala/endpoints4s/xhr/EndpointsSettings.scala

```diff
@@ -3,11 +3,12 @@ package endpoints4s.xhr
 import endpoints4s.Hashing
 
 import scala.annotation.nowarn
+import scala.concurrent.duration.{FiniteDuration}
 
 /** Settings for XHR interpreter.
   * @param baseUri Base of the URI of the service that implements the endpoints, can be absolute or relative (e.g. "http://foo.com" or "/bar")
   */
-final class EndpointsSettings private (val baseUri: Option[String]) extends Serializable {
+final class EndpointsSettings private (val baseUri: Option[String], val timeout: Option[FiniteDuration]) extends Serializable {
```

Comment:
Could you please also update the implementation of `hashCode`, `equals`, and `toString` to take into account the `timeout`?